### ISSUE TITLE
AttributeError: 'int' object has no attribute 'status'

### ISSUE THREAD

**Issue by asqui** [opsgenie/opsgenie-python-sdk#50]
New failure mode that might also be arising from unexpected responses from the REST API (hard to tell for certain because the inner exception detail is lost when it gets re-raised -- perhaps that is a separate issue that needs attention?)
```
  File "/opt/ecn/python_3rd_party/opsgenie_sdk/api/alert/__init__.py", line 2166, in list_alerts
    (data) = self.list_alerts_with_http_info(**kwargs)  # noqa: E501
  File "/opt/ecn/python_3rd_party/opsgenie_sdk/api/alert/__init__.py", line 2261, in list_alerts_with_http_info
    collection_formats=collection_formats)
  File "/opt/ecn/python_3rd_party/opsgenie_sdk/api_client.py", line 432, in call_api
    _preload_content, _request_timeout, _host)
  File "/opt/ecn/python_3rd_party/opsgenie_sdk/api_client.py", line 222, in __call_api
    raise exception
AttributeError: 'int' object has no attribute 'status'
```

This error happened only once and then went away, so that supports the hypothesis that it was some sort of strange REST response that caused it. If so, this will be the third recent example of API failure due to unexpected REST responses.

Maybe what is needed is a holistic approach to processing REST responses that triggers retries if any expectation about a normal REST response is not met?

**Comment by shilobar** [comment]
I'm seeing a similar exception several times a day:

```
  File "/usr/local/lib/python3.7/site-packages/opsgenie_sdk/api/alert/__init__.py", line 1136, in create_alert
    (data) = self.create_alert_with_http_info(create_alert_payload, **kwargs)  # noqa: E501
  File "/usr/local/lib/python3.7/site-packages/opsgenie_sdk/api/alert/__init__.py", line 1211, in create_alert_with_http_info
    collection_formats=collection_formats)
  File "/usr/local/lib/python3.7/site-packages/opsgenie_sdk/api_client.py", line 428, in call_api
    _preload_content, _request_timeout, _host)
  File "/usr/local/lib/python3.7/site-packages/opsgenie_sdk/api_client.py", line 218, in __call_api
    raise exception
  File "/usr/local/lib/python3.7/site-packages/opsgenie_sdk/api_client.py", line 202, in __call_api
    _request_timeout=_request_timeout)
  File "/usr/local/lib/python3.7/site-packages/tenacity/__init__.py", line 430, in __call__
    do = self.iter(retry_state=retry_state)
  File "/usr/local/lib/python3.7/site-packages/tenacity/__init__.py", line 367, in iter
    return fut.result()
  File "/usr/local/lib/python3.7/concurrent/futures/_base.py", line 428, in result
    return self.__get_result()
  File "/usr/local/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.7/site-packages/tenacity/__init__.py", line 433, in __call__
    result = fn(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/opsgenie_sdk/api_client.py", line 473, in request
    body=body)
  File "/usr/local/lib/python3.7/site-packages/opsgenie_sdk/rest.py", line 313, in POST
    body=body)
  File "/usr/local/lib/python3.7/site-packages/opsgenie_sdk/rest.py", line 264, in request
    exception = exceptions.build_exception(response=r)
  File "/usr/local/lib/python3.7/site-packages/opsgenie_sdk/exceptions.py", line 237, in build_exception
    exception.status = response.status
AttributeError: 'int' object has no attribute 'status'
```

**Comment by mhamzak008** [comment]
Hi @shilobar which version of the SDK are you using?

**Comment by shilobar** [comment]
@mhamzak008 I'm using `opsgenie-sdk==2.1.3`

**Comment by asqui** [comment]
This looks to be due to a bug in the implementation of `build_exception()` where it is doing a dictionary lookup to find an exception instance, but the default return value is the integer `-100`.

I guess what was intended was to have `ApiException()` as the default value.

https://github.com/opsgenie/opsgenie-python-sdk/blob/bb718aeea973bd758d63295954f82ce2a6006ed6/opsgenie_sdk/exceptions.py#L211-L242

**Comment by asqui** [comment]
I think this is now resolved by PR #55
